generators/artifacthub/scanner.go

```diff
@@ -20,18 +21,30 @@ const AhTextSearchQueryFieldName = "ts_query_web"
 
 func GetAhPackagesWithName(name string) ([]AhPackage, error) {
 	pkgs := make([]AhPackage, 0)
-	url := fmt.Sprintf("%s/packages/search?%s=%s&", ArtifactHubAPIEndpint, AhTextSearchQueryFieldName, name)
-	// add params
+
+	// Construct URL with encoded query parameters
+	baseURL, err := url.Parse(fmt.Sprintf("%s/packages/search", ArtifactHubAPIEndpint))
```

Comment:
```suggestion
	baseURL, err := url.Parse(fmt.Sprintf("%s/packages/search", ArtifactHubAPIEndpoint))
```
Will you fix the typo at the rest of the places?